src/lib/tecs/TECS.cpp

```diff
@@ -320,9 +323,11 @@ float TECSControl::_calcAirspeedControlOutput(const Setpoint &setpoint, const In
 	// if airspeed measurement is not enabled then always set the rate setpoint to zero in order to avoid constant rate setpoints
 	if (flag.airspeed_enabled) {
 		// Calculate limits for the demanded rate of change of speed based on physical performance limits
-		// with a 50% margin to allow the total energy controller to correct for errors.
-		const float max_tas_rate_sp = 0.5f * limit.STE_rate_max / math::max(input.tas, FLT_EPSILON);
-		const float min_tas_rate_sp = 0.5f * limit.STE_rate_min / math::max(input.tas, FLT_EPSILON);
+		// with a 50% margin to allow the total energy controller to correct for errors. Increase it in case of fast descend
+		const float max_tas_rate_sp = (param.fast_descend * 0.5f + 0.5f) * limit.STE_rate_max / math::max(input.tas,
```

Comment:
@KonradRudin Is there some rationale behind this calculation that I did not understand yet?

Reply:
So here, the energy change from the airspeed error is limited such that for the pitch controller, there is always a margin of 50% to correct for altitude. Since in fast descend, we only control the airspeed over the pitch control and not the altitude, it makes no sense to limit the energy change from airspeed to only 50% of the maximum allowed. It should be able to go up to 100%

Reply:
@KonradRudin Thanks, makes sense